src/main/java/com/github/coleb1911/ghost2/commands/modules/fun/ModuleCalculator.java

```diff
@@ -0,0 +1,84 @@
+package com.github.coleb1911.ghost2.commands.modules.fun;
+
+import com.github.coleb1911.ghost2.commands.meta.CommandContext;
+import com.github.coleb1911.ghost2.commands.meta.Module;
+import com.github.coleb1911.ghost2.commands.meta.ModuleInfo;
+import com.github.coleb1911.ghost2.commands.meta.ReflectiveAccess;
+import org.springframework.stereotype.Service;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * @author chicago4550
+ */
+
+
+
+@Service
+public final class ModuleCalculator extends Module {
+
+  @ReflectiveAccess
+  public ModuleCalculator(){
+    super(new ModuleInfo.Builder(ModuleCalculator.class)
+            .withName("Calculator")
+            .withAliases("calc")
+            .withDescription("Complete a 2-number calculation with either addition, subtraction, multiplication, or division"));
+
+  }
+
+    /**
+     * Constructs a new Module. Should only be used by subclasses of Module.
+     *
+     * @param info ModuleInfo.Builder with the subclass' metadata
+     */
+    protected ModuleCalculator(ModuleInfo.Builder info) {
+        super(info);
+    }
+
+    @Override
+    @ReflectiveAccess
+    public void invoke(@NotNull CommandContext ctx) {
+
+
+        List<String> Args = ctx.getArgs();
+
+        if (Args.size() != 3) {
+            ctx.getChannel().createMessage("this calculation is not in the correct format, please follow the format of (number calculation number)");
+        }
+
+        boolean incorrectInput = false;
+        for (int i = 0; i < Args.size(); i++) {
+            if (i % 2 == 0 && !incorrectInput) {
+                try {
+                    double f = Double.parseDouble(Args.get(i));
+                } catch (Exception e) {
+                    ctx.getChannel().createMessage("You entered an invalid number, please try this command again with a valid number");
+                    incorrectInput = true;
+                }
+            }
+
+        }
```

Comment:
Using a loop here is overengineering it and is less readable -- just fetch the operands by index and try to parse them.